sbi/src/api/attestation.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2022 by Rivos Inc.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::ecall_send;
+use crate::{AttestationFunction, Error, Result, SbiMessage};
+
+/// The data blob passed to the GetEvidence call must be 64 bytes long.
+pub const REQUEST_DATA_SIZE: usize = 64;
+
+/// Attestation evidence formats.
+#[derive(Copy, Clone, Debug)]
+#[repr(u64)]
+pub enum EvidenceFormat {
+    /// Single layer DICE TCB
+    /// https://trustedcomputinggroup.org/resource/dice-attestation-architecture/
+    DiceTcbInfo = 0,
+
+    /// X.509 extension for multiple layers DICE TCB.
+    /// https://trustedcomputinggroup.org/resource/dice-attestation-architecture/
+    DiceMultiTcbInfo = 1,
+
+    /// Open DICE profile
+    /// https://pigweed.googlesource.com/open-dice/+/HEAD/docs/specification.md
+    OpenDice = 2,
+}
+
+/// Get an attestation evidence.
+/// This function returns a X.509 certificate and the attestation evidence is
+/// included as an extension.
+///
+/// # Arguments
+///
+/// * `cert_request` - The [Certificate Signing Request]((https://datatracker.ietf.org/doc/html/rfc2986) buffer.
+/// * `request_data` - A data blob that will be included in the generated
+///                    certificate, as [UserNotice]((https://datatracker.ietf.org/doc/html/rfc2986)
+///                    X.509 certificate extension. This is typically used to
+///                    pass a cryptographic nonce.
+/// * `evidence_format` - The format of the attestation evidence as defined by [`EvidenceFormat`](crate::api::attestation::EvidenceFormat).
+/// * `cert_out` - The generated X.509 certificate buffer.
+pub fn get_evidence<'a>(
+    cert_request: &'a [u8],
+    request_data: &'a [u8],
+    evidence_format: EvidenceFormat,
+    cert_out: &'a mut [u8],
+) -> Result<u64> {
```

Comment:
What's the return value? If it's always 0 on success, would `Result<()>` be a more appropriate return type?

Reply:
I changed the API to return the certificate buffer itself. It makes it for a nicer high level SBI API.